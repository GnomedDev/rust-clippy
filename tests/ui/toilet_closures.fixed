#![warn(clippy::toilet_closures)]

fn main() {
    // Should lint
    let toliet: fn(u8) = drop;
    //~^ error: defined a 'toilet closure'
    let toliet_with_arg: fn(u8) = drop;
    //~^ error: defined a 'toilet closure'
    let toliet_with_typed_arg: fn(u8) = drop::<u8>;
    //~^ error: defined a 'toilet closure'
    let toliet_with_braces: fn(u8) = drop;
    //~^ error: defined a 'toilet closure'

    // Should not lint
    let toilet_higher_ranked: fn(&u8) = |_| ();
    let closure_multi_arg: fn(u8, u8) = |_, _| ();
    let closure_does_stuff: fn(u8) = |_| println!("Stuff!");
}
