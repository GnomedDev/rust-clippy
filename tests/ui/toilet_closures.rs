#![warn(clippy::toilet_closures)]

fn main() {
    // Should lint
    let toliet: fn(u8) = |_| ();
    //~^ error: defined a 'toilet closure'
    let toliet_with_arg: fn(u8) = |_to_drop| ();
    //~^ error: defined a 'toilet closure'
    let toliet_with_typed_arg: fn(u8) = |_to_drop: u8| ();
    //~^ error: defined a 'toilet closure'
    let toliet_with_braces: fn(u8) = |_| {};
    //~^ error: defined a 'toilet closure'

    // Should not lint
    let toilet_higher_ranked: fn(&u8) = |_| ();
    let closure_multi_arg: fn(u8, u8) = |_, _| ();
    let closure_does_stuff: fn(u8) = |_| println!("Stuff!");
}
