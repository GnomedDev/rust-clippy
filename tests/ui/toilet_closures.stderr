error: defined a 'toilet closure'
  --> tests/ui/toilet_closures.rs:5:26
   |
LL |     let toliet: fn(u8) = |_| ();
   |                          ^^^^^^ help: replace with: `drop`
   |
   = note: `-D clippy::toilet-closures` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::toilet_closures)]`

error: defined a 'toilet closure'
  --> tests/ui/toilet_closures.rs:7:35
   |
LL |     let toliet_with_arg: fn(u8) = |_to_drop| ();
   |                                   ^^^^^^^^^^^^^ help: replace with: `drop`

error: defined a 'toilet closure'
  --> tests/ui/toilet_closures.rs:9:41
   |
LL |     let toliet_with_typed_arg: fn(u8) = |_to_drop: u8| ();
   |                                         ^^^^^^^^^^^^^^^^^ help: replace with: `drop::<u8>`

error: defined a 'toilet closure'
  --> tests/ui/toilet_closures.rs:11:38
   |
LL |     let toliet_with_braces: fn(u8) = |_| {};
   |                                      ^^^^^^ help: replace with: `drop`

error: aborting due to 4 previous errors

